src/init.rs

```diff
@@ -62,15 +80,18 @@ pub fn init_stub(shell_name: &str) {
     if let Some(script) = setup_stub {
         print!("{}", script);
     };
+    Ok(())
 }
 
 /* This function (called when `--print-full-init` is passed to `starship init`)
 prints out the main initialization script */
-pub fn init_main(shell_name: &str) {
+pub fn init_main(shell_name: &str) -> io::Result<()> {
+    let starship_path = get_starship_path()?.replace("\"", "\"\\\"\"");
+
     let setup_script = match shell_name {
-        "bash" => Some(BASH_INIT),
-        "zsh" => Some(ZSH_INIT),
-        "fish" => Some(FISH_INIT),
+        "bash" => Some(BASH_INIT.replace("STARSHIP_EXEC", &format!("\"{}\"", starship_path))),
```

Comment:
Smart :clap: . I had it in my head for the longest time to use `lazy_static!` for this purpose, but this is clearly much cleaner.

Reply:
Thanks ☺️